### PR TITLE
Adds the 2 lessons that cover one to many relationships.

### PR DESCRIPTION
**Pull request by KatBrandt**
# Curriculum PR Template

### Does this PR close any issues?
#99 

### Description
This PR pulls in two lessons, both of which are needed to cover the one-to-many relationships, at the db level, and in the rails application.
- Since Model Testing hasn't been covered yet, I added a bit more to this lesson about model testing. I do not think another lesson on model testing is necessary. 

### What questions do you have/what do you want feedback on? (optional)
- I was thinking the one-to-many-relationships(database level) would be part I (either as homework or in a class session) and then one-to-many-relationships(rails level) would be part II, to be done in class. Does this sound reasonable? 
- This combo of lessons is definitely big. It's all one big topic though, so not sure we could or would want to break it up even more. 
Open to thoughts and ideas!

### CODE REVIEW

**Review by jamisonordway — COMMENTED** [review]
I agree that it makes sense to break this into 2 lessons, one focused on DB and another focused on the Rails side. 

I also like that model testing is included here; it's pretty similar to the unit testing they've done in mod 1, so this is probably a good opportunity to make space on the calendar for something else. By making it part of AR associations lesson, it feels like intentionally incorporated TDD.

File: module2/lessons/one_to_many_relationships_part1.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+* During this week, we will start using an application called Set List in class, which will have a songs database table. The table will store songs' title, length and seconds, and its play count. If we wanted to store and relate songs to an artist (for example, the artist Prince has two songs in our app: 'Purple Rain' and 'Raspberry Beret') how might you try to incorporate that data into the Set List database? Sketch out an example. 
+
+## What is a database?
+
+Before we jump into how we might relate a song to a specific artist, let's take a step back to review what a database *is*. At the highest level, a database is a location where we can store information and allow that information to persist in our application. Right now, in our app, we have a database that is storing song information. More specifically, we have a `database` that has a songs `table` and that table has columns that store each song's `attributes`; we could visualize this configuration like this:
+
+<p align='center'>
+  <img src='./images/1_to_many_db_image_1.png'>
+</p>
```
[the remark this answers is not in the document]
Comment (zoefarrell) [reply module2/lessons/one_to_many_relationships_part1.md:31]: Good call about the images! It looks like there is a pattern started from the `html` lesson to add a folder within `assets/images/` for each lesson.

**Comment by zoefarrell** [comment]
This PR looks good! I agree that the two parts make sense and the first part can be async. The second lesson feels quite large so will probably want to find a full 3-hour chunk on the calendar.

@KatBrandt can you add in those images and then let's merge!

File: module2/lessons/one_to_many_relationships_part1.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (KatBrandt) [reply module2/lessons/one_to_many_relationships_part1.md:31]: Perfect! Got these added.

**Pull request merged**